test/common/http/conn_manager_impl_test.cc

```diff
@@ -801,6 +801,89 @@ TEST_F(HttpConnectionManagerImplTest, RouteShouldUseNormalizedHost) {
   filter_callbacks_.connection_.raiseEvent(Network::ConnectionEvent::RemoteClose);
 }
 
+// Filters observe host header w/o trailing dot in host when trailing dot removal is configured
```

Comment:
Do you think dot stripping could be tested more directly? Instantiate `ConnectionManagerImpl`, create a `RequestDecoder` by invoking `ConnectionManagerImpl::newStream()`, then invoke `RequestDecoder::decodeHeaders()` and verify the contents of the "host" header. WDYT?

Reply:
yes, added test for this, please check and let me know. Thank you.

Reply:
Ultimately, I'm not sure why three tests that very closely resemble each other are needed (they all verify that a) header map is updated, and b) that filters in the filter chain see that change. Perhaps just `RemoveTrailingHostDot` or `FilterShouldUseHostWithoutTrailingDot` is sufficient?

Reply:
Agree, will remove `RemoveTrailingHostDot` test and add check to verify host header after `RequestDecoder::decodeHeaders()` in `FilterShouldUseHostWithoutTrailingDot` and `RouteShouldUseHostWithoutTrailingDot`.

Reply:
Addressed.